File: quanlysinhvien3 - springboot/src/main/java/com/example/quanlysinhvien/entity/KetQuaHocTap.java
package com.example.quanlysinhvien.entity;


import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.OnDelete;
import org.hibernate.annotations.OnDeleteAction;
import org.hibernate.annotations.UpdateTimestamp;

import java.time.LocalDateTime;

@Getter
@Setter
@Entity
@Table(name = "ket_qua_hoc_tap")
public class KetQuaHocTap {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    private Long id;

    @Column(name = "diem_qua_trinh")
    private Double diem_qua_trinh;


    @Column(name = "diem_thanh_phan")
    private Double diem_thanh_phan;

    @Column(name = "created_at", nullable = false, updatable = false)
    @CreationTimestamp
    private LocalDateTime createdAt;

    @Column(name = "update_at")
    @UpdateTimestamp
    private LocalDateTime updateAt;

  @ManyToOne
  @JoinColumn(name = "ma_sv",
          //unique = true
          nullable = false
  )
    @OnDelete(action = OnDeleteAction.CASCADE)
    private SinhVien sinhVien;

    @ManyToOne
    @JoinColumn(name = "ma_mon",
            //unique = true
            nullable = false
    )
    @OnDelete(action = OnDeleteAction.CASCADE)
    private MonHoc monHoc;









}

File: angular/qlsv/qlsv/src/app/models/ket-qua-hoc-tap.ts
export interface KetQuaHocTap {
  id: number;
  diem_qua_trinh: number;
  diem_thanh_phan: number;
  sinhVienTen: string;
  monHocTen: string;

  // --- THÊM 2 DÒNG NÀY ---
  sinhVienId: number;
  monHocId: number;

  // Các trường phụ (có thể có hoặc không)
  createdAt?: string;
  updatedAt?: string;
}
